src/Microsoft.Health.TaskManagement/JobHosting.cs

```diff
@@ -0,0 +1,250 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.JobManagement
+{
+    public class JobHosting
+    {
+        private IQueueClient _queueClient;
+        private IJobFactory _jobFactory;
+        private ILogger<JobHosting> _logger;
+        private ConcurrentDictionary<long, Func<Task>> _activeJobsNeedKeepAlive = new ConcurrentDictionary<long, Func<Task>>();
+
+        public JobHosting(IQueueClient queueClient, IJobFactory jobFactory, ILogger<JobHosting> logger)
+        {
+            EnsureArg.IsNotNull(queueClient, nameof(queueClient));
+            EnsureArg.IsNotNull(jobFactory, nameof(jobFactory));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _queueClient = queueClient;
+            _jobFactory = jobFactory;
+            _logger = logger;
+        }
+
+        public int PollingFrequencyInSeconds { get; set; } = Constants.DefaultPollingFrequencyInSeconds;
+
+        public short MaxRunningJobCount { get; set; } = Constants.DefaultMaxRunningJobCount;
+
+        public int JobHeartbeatTimeoutThresholdInSeconds { get; set; } = Constants.DefaultJobHeartbeatTimeoutThresholdInSeconds;
+
+        public int JobHeartbeatIntervalInSeconds { get; set; } = Constants.DefaultJobHeartbeatIntervalInSeconds;
+
+        public byte StartPartitionId { get; set; }
```

Comment:
>public byte StartPartitionId { get; set; } [](http://example.com/codeflow?start=8&length=42)

Please remove. We don't need this as it is handled inside dequeue stored procedure with no negative effects. In future we can remove this parameter from dequeue stored procedure completely.

Reply:
thanks, updated.

Reply:
I still see this in latest version. Please remove.